rig-sqlite/src/lib.rs

```diff
@@ -0,0 +1,360 @@
+                        );
+                        stmt.execute(rusqlite::params![doc_id, blob])
+                            .map_err(tokio_rusqlite::Error::from)?;
+                    }
+                }
+
+                tx.commit().map_err(tokio_rusqlite::Error::from)?;
+                Ok(())
+            })
+            .await
+            .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))?;
+
+        Ok(())
+    }
+
+    async fn get_document<T: for<'a> Deserialize<'a>>(
+        &self,
+        id: &str,
+    ) -> Result<Option<T>, VectorStoreError> {
+        debug!("Fetching document with id {}", id);
+        let id_clone = id.to_string();
+        let doc_str = self
+            .conn
+            .call(move |conn| {
+                conn.query_row(
+                    "SELECT document FROM documents WHERE doc_id = ?1",
+                    rusqlite::params![id_clone],
+                    |row| row.get::<_, String>(0),
+                )
+                .optional()
+                .map_err(tokio_rusqlite::Error::from)
+            })
+            .await
+            .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))?;
+
+        match doc_str {
+            Some(doc_str) => {
+                let doc: T = serde_json::from_str(&doc_str)
+                    .map_err(|e| VectorStoreError::DatastoreError(Box::new(e)))?;
+                Ok(Some(doc))
+            }
+            None => {
+                debug!("No document found with id {}", id);
+                Ok(None)
+            }
+        }
+    }
+
+    async fn get_document_embeddings(
+        &self,
+        id: &str,
+    ) -> Result<Option<DocumentEmbeddings>, VectorStoreError> {
+        debug!("Fetching embeddings for document {}", id);
+        let id_clone = id.to_string();
+        let result = self
+            .conn
+            .call(move |conn| {
+                let mut stmt = conn.prepare(
+                    "SELECT e.embedding, d.document 
+                     FROM embeddings e
+                     JOIN documents d ON e.rowid = d.id
+                     WHERE d.doc_id = ?1",
+                )?;
+
+                let result = stmt
+                    .query_map(rusqlite::params![id_clone], |row| {
+                        let bytes: Vec<u8> = row.get(0)?;
+                        let doc_str: String = row.get(1)?;
+                        let doc: serde_json::Value =
+                            serde_json::from_str(&doc_str).map_err(|e| {
+                                rusqlite::Error::FromSqlConversionFailure(
+                                    0,
+                                    rusqlite::types::Type::Text,
+                                    Box::new(e),
+                                )
+                            })?;
+                        let vec = bytes
+                            .chunks(4)
+                            .map(|chunk| f32::from_le_bytes(chunk.try_into().unwrap()) as f64)
```

Comment:
Rather than using `unwrap`, please opt to use `expect` so that a defined reason can be provided here (even if it's impossible). This makes it easier for future maintainers as it's self-documenting.